tendrl/commons/utils/ansible_module_runner.py

```diff
@@ -31,26 +31,42 @@ class AnsibleRunner(object):
 
     """
 
-    def __init__(self, module_path, **kwargs):
+    def __init__(
+        self,
+        module_path,
+        publisher_id=None,
+        node_id=None,
+        socket_path=None,
+        **kwargs
+    ):
         self.module_path = modules.__path__[0] + "/" + module_path
+        self.socket_path = socket_path
+        self.publisher_id = publisher_id
+        if not self.publisher_id:
+            self.publisher_id = NS.publisher_id
+        self.node_id = node_id
```

Comment:
what if node_id is None?

Use
self.node_id = node_id or NS.node_context.node_id